core/app/[locale]/(default)/(auth)/logout/route.ts

```diff
@@ -1,14 +1,19 @@
-import { getLocale } from 'next-intl/server';
+import { NextRequest } from 'next/server';
 
 import { signOut } from '~/auth';
 import { redirect } from '~/i18n/routing';
 import { setForceRefreshCookie } from '~/lib/force-refresh';
 
-export const GET = async () => {
-  const locale = await getLocale();
+export const GET = async (
+  request: NextRequest,
+  { params }: { params: Promise<{ locale: string }> },
+) => {
+  const { locale } = await params;
+  const redirectTo = request.nextUrl.searchParams.get('redirectTo') ?? '/login';
```

Comment:
Is there any security risk if we don't have an exclusive list of allowed redirect origins? Could an attacker attempt to craft a malicious URL by having logout redirect to a phishing site that looks otherwise like a login screen?

Reply:
Redirecting with URL search params is quite a common thing so I would imagine this isn't a concern but I wonder if usage of a nonce or some other piece of state is generally used to prevent spoofing and whether that is necessary in this context

Reply:
We are essentially doing the same thing on the `/login` path and had @bc-donfran review that path https://github.com/bigcommerce/catalyst/pull/2182#discussion_r2023549260 and give his blessings from a security standpoint. I assumed it was fine to take the same approach here and only accept a relative URL.

Now that I look more closely, I do see a miss on my end – I forgot to do `.pathname` after the `new URL` so I'll push a fix for that.

Reply:
https://github.com/bigcommerce/catalyst/pull/2198